pkg/machinepool/machinepool_test.go

```diff
@@ -26,6 +28,107 @@ var _ = Describe("Machinepool and nodepool", func() {
 				Value(cmv1.UpgradePolicyStateValueScheduled)).
 				NextRun(date)
 		})
+		It("editAutoscaling should equal nil if nothing is changed", func() {
+			nodepool, err := cmv1.NewNodePool().
+				Autoscaling(cmv1.NewNodePoolAutoscaling().MaxReplica(2).MinReplica(1)).
+				Build()
+			Expect(err).ToNot(HaveOccurred())
+			builder := editAutoscaling(nodepool, 1, 2)
+			Expect(builder).To(BeNil())
+		})
+		It("editAutoscaling should equal the exepcted output", func() {
+			nodepool, err := cmv1.NewNodePool().
+				Autoscaling(cmv1.NewNodePoolAutoscaling().MaxReplica(2).MinReplica(1)).
+				Build()
+			Expect(err).ToNot(HaveOccurred())
+			builder := editAutoscaling(nodepool, 2, 3)
+			asBuilder := cmv1.NewNodePoolAutoscaling().MaxReplica(3).MinReplica(2)
+			Expect(builder).To(Equal(asBuilder))
+		})
+		It("editAutoscaling should equal the exepcted output with no min replica value", func() {
+			nodepool, err := cmv1.NewNodePool().
+				Autoscaling(cmv1.NewNodePoolAutoscaling().MaxReplica(2).MinReplica(1)).
+				Build()
+			Expect(err).ToNot(HaveOccurred())
+			builder := editAutoscaling(nodepool, 0, 3)
+			asBuilder := cmv1.NewNodePoolAutoscaling().MaxReplica(3).MinReplica(1)
+			Expect(builder).To(Equal(asBuilder))
+		})
+
+		It("editAutoscaling should equal the exepcted output with no max replica value", func() {
+			nodepool, err := cmv1.NewNodePool().
+				Autoscaling(cmv1.NewNodePoolAutoscaling().MaxReplica(4).MinReplica(1)).
+				Build()
+			Expect(err).ToNot(HaveOccurred())
+			builder := editAutoscaling(nodepool, 2, 0)
+			asBuilder := cmv1.NewNodePoolAutoscaling().MaxReplica(4).MinReplica(2)
+			Expect(builder).To(Equal(asBuilder))
+		})
+		Context("Prompt For NodePoolNodeRecreate", func() {
+			var mockPromptFuncInvoked bool
+			var t *TestingRuntime
+			BeforeEach(func() {
+				t = NewTestRuntime()
+				mockPromptFuncInvoked = false
+			})
+
+			It("Prompts when the user has deleted a kubelet-config", func() {
+
+				invoked := func(r *rosa.Runtime) bool {
+					mockPromptFuncInvoked = true
+					return mockPromptFuncInvoked
+				}
+
+				original := MockNodePool(func(n *cmv1.NodePoolBuilder) {
+					n.KubeletConfigs("test")
+				})
+
+				update := MockNodePool(func(n *cmv1.NodePoolBuilder) {
+					n.KubeletConfigs("")
+				})
+
+				Expect(promptForNodePoolNodeRecreate(original, update, invoked, t.RosaRuntime)).To(BeTrue())
+				Expect(mockPromptFuncInvoked).To(BeTrue())
+			})
```

Comment:
DRY
The
```go
				invoked := func(r *rosa.Runtime) bool {
					mockPromptFuncInvoked = true
					return mockPromptFuncInvoked
				}
```
does not need to be repeated in each test

```suggestion
		Context("Prompt For NodePoolNodeRecreate", func() {
			var mockPromptFuncInvoked bool
			var t *TestingRuntime
			BeforeEach(func() {
				t = NewTestRuntime()
				mockPromptFuncInvoked = false
			})

			mockPromptFunc := func(r *rosa.Runtime) bool {
				mockPromptFuncInvoked = true
				return mockPromptFuncInvoked
			}


			It("Prompts when the user has deleted a kubelet-config", func() {
				original := MockNodePool(func(n *cmv1.NodePoolBuilder) {
					n.KubeletConfigs("test")
				})

				update := MockNodePool(func(n *cmv1.NodePoolBuilder) {
					n.KubeletConfigs("")
				})

				Expect(promptForNodePoolNodeRecreate(original, update, mockPromptFunc, t.RosaRuntime)).To(BeTrue())
				Expect(mockPromptFuncInvoked).To(BeTrue())
			})
```
The same `mockPromptFunc` can be used in each test.